perf/perf_sdt_probe.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2018 IBM.
+# Author: Kamalesh Babulal <kamalesh@linux.vnet.ibm.com>
+# Modified And tested : Praveen K Pandey <praveen@linux.vnet.ibm.com>
+
+import os
+import platform
+from avocado import Test
+from avocado import main
+from avocado.utils import distro
+from avocado.utils import process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class PerfSDT(Test):
+
+    """
+    Test userspace SDT markers
+    :avocado: tags=privileged
+    """
+
+    def run_cmd(self, cmd):
+        if process.system(cmd, ignore_status=True, sudo=True, shell=True):
+            self.is_fail += 1
+        return
+
+    @staticmethod
+    def run_cmd_out(cmd):
+        return process.system_output(cmd, shell=True,
+                                     ignore_status=True, sudo=True)
+
+    def add_library(self):
+        """
+        Find the libpthread path, it differs for distros
+        example:
+        libpthread.so.0 (libc6,64bit, OS ABI: Linux 3.10.0) =>
+            /lib/powerpc64le-linux-gnu/libpthread.so.0
+        """
+        self.libpthread = self.run_cmd_out("ldconfig -p|grep -i "
+                                           "libpthread|grep 64").split(" ")[7]
+        if not self.libpthread:
+            self.fail("Library %s not found", self.libpthread)
+
+        val = self.run_cmd_out("perf list | grep SDT|wc -l")
+        # Add the libpthread.so.0 to perf
+        perf_add = "perf buildid-cache -v --add %s" % self.libpthread
+        self.is_fail = 0
+        self.run_cmd(perf_add)
+        if self.is_fail:
+            self.fail("Unable to add %s to builid-cache", self.libpthread)
+
+        # Check if libpthread has valid SDT markers
+        new_val = self.run_cmd_out("perf list | grep SDT|wc -l")
```

Comment:
better to modify into native python in this commit itself.

Reply:
sure let me see how i can do that